src/System.Net.Utilities/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -2,33 +2,268 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.ComponentModel;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
 using System.Threading;
+using System.Diagnostics;
+using System.Text;
+using System.IO;
 
 namespace System.Net.NetworkInformation
 {
     public partial class Ping
     {
-        private void InternalSendAsync(IPAddress address, byte[] buffer, int timeout, PingOptions options)
+        private const int IcmpHeaderLengthInBytes = 8;
+        private const int IpHeaderLengthInBytes = 20;
+
+        private static readonly bool s_canUseRawSockets = CheckRawSocketPermissions();
+
+        // Ubuntu has ping under /bin, OSX under /sbin.
+        private static readonly string[] s_binFolders = { "/bin", "/sbin" };
```

Comment:
I wasn't aware we probed through the PATH variable, I thought we just looked in the current directory. That makes things easier for this, then.